utils/interfaces/_core.py

```diff
@@ -170,14 +170,15 @@ def get_rid_from_span(span):
         return None
 
     meta = span.get("meta", {})
+    metrics = span.get("metrics", {})
 
     user_agent = None
 
     if span.get("type") == "rpc":
         user_agent = meta.get("grpc.metadata.user-agent")
         # java does not fill this tag; it uses the normal http tags
 
-    if not user_agent:
+    if not user_agent and metrics.get("_dd.top_level") == 1.0:
```

Comment:
It would be good to add comment explaining the new check

Reply:
This kind of works, but it feels indirect:

- there can be multiple top level tags in the trace (e.g with request queuing synthetic spans, which are in fact at the root of the trace). we don't run in this way but we should have weblog cases for that situation.
- if this is squarely looking for the appsec case then `_dd.appsec.enabled =1.0` is a better, unambiguous marker
- it also begs into question why appsec and tracing duplicate tagging in different places

That said, I'd rather have the CI green again, so OK to merge and fix later while we think about it (cc @marcotc)